pyo3-macros-backend/src/pyclass.rs

```diff
@@ -1162,7 +1178,15 @@ fn complex_enum_struct_variant_new<'a>(
         }
         args
     };
-    let signature = crate::pyfunction::FunctionSignature::from_arguments(args)?;
+
+    let signature = if let Some(constructor) = variant.options.constructor {
+        crate::pyfunction::FunctionSignature::from_arguments_and_attribute(
+            args,
+            constructor.into_signature(),
+        )?
+    } else {
+        crate::pyfunction::FunctionSignature::from_arguments(args)?
+    };
```

Comment:
😂 so nice to see that this became almost trivial logic-wise!

Reply:
Yes! I took a few tries, because I needed the signature by value instead of by reference only, but it worked out to be pretty simple in the end.